osquery/tables/system/darwin/apps.mm

```diff
@@ -184,6 +184,24 @@ void genApplication(const pt::ptree& tree,
   r["name"] = path.parent_path().parent_path().filename().string();
   r["path"] = path.parent_path().parent_path().string();
 
+  NSString* filePath =
+      [NSString stringWithUTF8String:path.parent_path().parent_path().c_str()];
+  MDItemRef mdItem = MDItemCreate(NULL, (CFStringRef)filePath);
+
+  if (mdItem != NULL) {
+    NSDate* lastOpened = (NSDate*)CFBridgingRelease(
+        MDItemCopyAttribute(mdItem, kMDItemLastUsedDate));
+    if (lastOpened != NULL) {
+      r["last_opened_time"] = INTEGER([lastOpened timeIntervalSince1970]);
+      CFRelease(mdItem);
```

Comment:
This seems misplaced. From line `191` if `mdItem` is allocated, then the branch taken in line `198` where `lastOpened` is `NULL` will not release `mdItem`, leading to a leak.